pkg/reporter/reporter.go

```diff
@@ -481,3 +516,40 @@ func wgWait(ctx context.Context, processName string, maxRoutines int, done chan
 
 	done <- true
 }
+
+func getLabelsFromMetricQuery(queryLabel string, originalQueryString string) (parsedLabels string) {
+	arr := strings.Split(originalQueryString, "(")
+	for _, word := range arr {
+		if strings.Contains(word, queryLabel) {
+			parsedLabels = utils.GetStringInBetween(originalQueryString, "{", "}")
+		}
+	}
+
+	return parsedLabels
+}
+
+func addAdditionalFieldKVPairsToMap(labelPairs []string, additionalField string ,additionalFieldMap map[string]string){
+	for _, keyValuePairString := range labelPairs {
+		keyValuePairArray := strings.Split(keyValuePairString, "=")
+		if additionalField == keyValuePairArray[0] {
+			additionalFieldMap[keyValuePairArray[0]] = keyValuePairArray[1]
+		}
+	}
+}
+
+func processAdditionalFields(queryLabel string, originalQueryString string, additionalFieldsFromMeterDef []string, additionalFieldMap map[string]string) []error {
```

Comment:
This function is setting the value off of the query. We want the additional field to be carried over through the query, not necessarily set off of the query. You would add the field as an additionalField to the downstream so it can be pulled out of the result.